app/market/item/[id]/page.jsx

```diff
@@ -1,43 +1,136 @@
 "use client";
-
-import { useEffect, useState } from "react";
+import * as React from "react";
+import { Box, Button, Typography, Card, CardActionArea, CardMedia, Breadcrumbs, Link, ThemeProvider } from "@mui/material";
 import { useRouter } from "next/navigation";
-import { Box, Button } from "@mui/material";
-import CardComponent from "@/components/CardComponent";
-
-/**
- * @param {params} Object
- 
-*/
-export default function Page({ params }) {
-  const [data, setData] = useState(null);
+import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';
+import { theme } from "@/styles/theme";
+
+// Function to convert currency code to symbol
+const getCurrencySymbol = (currencyCode) => {
+  const currencySymbols = {
+    USD: "$",
+    EUR: "€",
+    GBP: "£",
+  };
+  return currencySymbols[currencyCode] || currencyCode;
+};
+
+const IndividualCardPage = ({ params }) => {
+  const [cardDetails, setCardDetails] = React.useState(null);
   const router = useRouter();
-  const id = params.id;
-  useEffect(() => {
-    //fetch data need to move to file in utils
-    const fetchCard = async () => {
-      const response = await fetch(`/api/cards/${id}`);
-      if (!response.ok) {
-        throw new Error(response.statusText);
+  const { id } = params;
+
+  React.useEffect(() => {
+    const fetchCardDetails = async () => {
+      try {
+        const response = await fetch(`/api/cards/${id}`);
+        if (!response.ok) {
+          throw new Error('Failed to fetch card details');
+        }
+        const data = await response.json();
+        setCardDetails(data.data); // Update to set the entire data object
+      } catch (error) {
+        console.error('Error fetching card details:', error);
+        setCardDetails(null);
       }
-      const data = await response.json();
-      setData(data.data);
     };
-    fetchCard();
+
+    if (id) {
+      fetchCardDetails();
+    }
   }, [id]);
 
+  const handleEditButtonClick = () => {
+    router.push(`/sell/edit/${id}`);
+  };
+
+  const handleSellerInfoButtonClick = () => {
+    router.push(`/market/seller/${id}`);
```

Comment:
this id - it is the card id, not seller. Maybe it would be better if we will leave it like /market/seller/[id] and add comment that we will add this route later?